Source/FakeItEasy/Creation/FakeOptions.cs

```diff
@@ -20,9 +22,19 @@ public static FakeOptions Empty
             get { return new FakeOptions(); }
         }
 
-        public object WrappedInstance { get; set; }
+        public DefaultFakeWrapperConfigurer Wrapper
+        {
+            get
+            {
+                return this.wrapper;
+            }
 
-        public ISelfInitializingFakeRecorder SelfInitializedFakeRecorder { get; set; }
+            set
+            {
+                this.wrapper = value;
+                this.FakeConfigurationActions.Add(fake => this.wrapper.ConfigureFakeToWrap(fake));

```

Comment:
I wonder if we can come up with a better shape for this class. In it's current form, setting this property obviously has a side effect on `FakeConfigurationActions` which itself has a property setter.